pkg/remote/aws/route53_health_check_supplier_test.go

```diff
@@ -0,0 +1,93 @@
+package aws
+
+import (
+	"context"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/aws-sdk-go/service/route53"
+
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/cloudskiff/driftctl/mocks"
+	"github.com/cloudskiff/driftctl/pkg/parallel"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+	"github.com/cloudskiff/driftctl/test"
+	"github.com/cloudskiff/driftctl/test/goldenfile"
+	mocks2 "github.com/cloudskiff/driftctl/test/mocks"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	resourceaws "github.com/cloudskiff/driftctl/pkg/resource/aws"
+	awsdeserializer "github.com/cloudskiff/driftctl/pkg/resource/aws/deserializer"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+)
+
+func TestRoute53HealthCheckSupplier_Resources(t *testing.T) {
+	cases := []struct {
+		test    string
+		dirName string
+		mocks   func(client *mocks.Route53Repository)
+		err     error
+	}{
+		{
+			test:    "no health check",
+			dirName: "route53_health_check_empty",
+			mocks: func(client *mocks.Route53Repository) {
+				client.On("ListAllHealthChecks").Return([]*route53.HealthCheck{}, nil)
+			},
+			err: nil,
+		},
+		{
+			test:    "Multiple health check",
+			dirName: "route53_health_check_multiple",
+			mocks: func(client *mocks.Route53Repository) {
+				client.On("ListAllHealthChecks").Return([]*route53.HealthCheck{
+					{Id: aws.String("43c6ded9-1978-46cb-aa1f-e38a899076da")},
+					{Id: aws.String("8b679eaa-b93b-41b2-b406-dac956f2d20d")},
+				}, nil)
+			},
+			err: nil,
+		},
+		{
+			test:    "cannot list health check",
+			dirName: "route53_health_check_list_error",
```

Comment:
Do we want to keep 2 empty results since the first test will have the same result ?

Reply:
We are testing error behaviour there it's not the same think that testing no results are returned from AWS side

Reply:
From the product point of view it is indeed not the same, but as of the mocked golden file reader we return a slice of `[]resource.Resource` which here is an empty array.

I can easily see a future where we could want to test other errors and where with this logic the number of useless results can only increases exponentially.

Reply:
I don't get your point, we need this test to test theses lines of code :

![image](https://user-images.githubusercontent.com/6154987/107971957-6c325880-6fb3-11eb-9e54-d797a896a793.png)

Reply:
You could test this line like this:

```go
{
  test:    "cannot list health check",
  dirName: "route53_health_check_empty",
  mocks: func(client *mocks.Route53Repository) {
    client.On("ListAllHealthChecks").Return(nil, awserr.NewRequestFailure(nil, 403, ""))
  },
  err: remoteerror.NewResourceEnumerationError(awserr.NewRequestFailure(nil, 403, ""), resourceaws.AwsRoute53HealthCheckResourceType),
}
```

Look at the `dirName` here. I reused the one with empty results so that we don't need to create another empty results but with different naming. I did it like that in SQS.

Reply:
Oh ok I got your point ! We can totally do this good idea 👍🏻

Reply:
yeah the only matter is that when we do that we cannot use -update to generate the folder so I used different folders in all the enumeration error test on all the other resources to keep -update behavior.

Reply:
May I suggest that we stop at the error testing if it fail so we use no folder at all for error testing ? it's not really useful to test the list values and it's not mandatory that this return an empty list.

Reply:
ok that cannot work as we need the schema... I can use the same folder but it's not what I've done in other resource and I'm not a big fan a generating the golden file for a test and using them for another. We should find a way to have only one schema if we want to save space on golden files

Reply:
ok it cannot work as we need the schema. I think we should find a way to use only one schema if we want to save space I'm not a really big fan of generating golden file for a test and using them for another...

Reply:
I'm good with that, we should find a way to reuse schema

Reply:
Ok then let's break it into 2 PRs, this one and mine on cloudfront will still have this kind of different folder with its own results and schema. Another PR will deal with a new architecture where we use only one schema per resource. Are you ok with that ?

Reply:
for now i reused empty test folder like you asked, But yeah we should do a dedicated pr to refactor this